src/lib/common/include/sol-types.h

```diff
@@ -242,6 +242,12 @@ struct sol_blob_type {
  * The default type uses free() to release the blob's memory
  */
 extern const struct sol_blob_type *SOL_BLOB_TYPE_DEFAULT;
+/*
+ * The no-free type doesn't free blob's memory. Used when pointing to inner
+ * position of a pre existing blob or any other case when blob's memory
+ * shouldn't be freed
```

Comment:
hum... this may cause a confusion. The blob `data` memory (internal pointed) is not freed. The `struct sol_blob` itself will be freed. Would try to clarify it.